app/forms/register/company/DirectorContactDetailsFormProvider.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.register.company
+
+import forms.mappings.Mappings
+import javax.inject.Inject
+import models.register.company.DirectorContactDetails
+import play.api.data.Form
+import play.api.data.Forms._
+
+class DirectorContactDetailsFormProvider @Inject() extends Mappings {
+
+   def apply(): Form[DirectorContactDetails] = Form(
+     mapping(
+      "field1" -> text("directorContactDetails.error.field1.required")
+        .verifying(maxLength(132, "directorContactDetails.error.field1.length")),
+      "field2" -> text("directorContactDetails.error.field2.required")
+        .verifying(maxLength(24, "directorContactDetails.error.field2.length"))
+    )(DirectorContactDetails.apply)(DirectorContactDetails.unapply)
+   )
+ }
```

Comment:
Could we use email address and phone number rather than field1 and field2? Also there is no regex validation.